homeassistant/components/starlink/coordinator.py

```diff
@@ -0,0 +1,38 @@
+"""Contains the shared Coordinator for Starlink systems."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from starlink_grpc import ChannelContext, GrpcError, StatusDict, status_data
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class StarlinkUpdateCoordinator(DataUpdateCoordinator[StatusDict]):
+    """Coordinates updates between all Starlink sensors defined in this file."""
+
+    def __init__(self, hass: HomeAssistant, name: str, url: str) -> None:
+        """Initialize an UpdateCoordinator for a group of sensors."""
+        self.channel_context = ChannelContext(target=url)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=1),
+        )
+
+    async def _async_update_data(self) -> StatusDict:
+        async with async_timeout.timeout(1):
```

Comment:
```suggestion
        async with async_timeout.timeout(4):
```

1 seems like a low timeout here.  If the event loop is busy it going to fail.

Reply:
Would it make more sense to move the timeout into the `add_executor_job` lambda in this case?

Reply:
If the timeout is happening in the thread you'll effectively get the full timeout.  If you use `async_timeout` its going to include the time it takes for the event loop to pick up the job, hand it off to the executor and get the response back.

Reply:
Ah ok, so not necessary here. Thanks!